### PR TITLE
version 1 

### PR DESCRIPTION
**Pull request by ethenotethan**

### CODE REVIEW

File: src/indicators.pyx [hunk 1]
```diff
@@ -92,17 +94,88 @@ cpdef float WMA(double[:] closes, int period, float weightedFactor):
...
+        assert(period>0, ("out of range, period='%s'" %period))
+
+        cdef float currentWeight
+        cdef float numerator      
+        cdef float denominator 
+        cdef int i 
+        cdef int j 
+        cdef int k = 0
+        cdef int length = closes.shape[0]
+        cdef double *results = <double *> malloc(sizeof(double)* (period+1))
```
Comment (adrain-cb) [inline src/indicators.pyx:137]: Sickk

File: src/indicators.pyx [hunk 2]
```diff
@@ -92,17 +94,88 @@ cpdef float WMA(double[:] closes, int period, float weightedFactor):
...
+                denominator = 0.0
+                for j in xrange(i, 0, -1):
+                        numerator += closes[j] * currentWeight
+                        denominator += currentWeight
+
+                        currentWeight *= alpha
+                        if currentWeight < epsilon:
+                                break
+                results[k] = numerator / denominator
+                k+=1
```
Comment (adrain-cb) [inline src/indicators.pyx]: You can do `for in` loops in cython 👀
If it works the same as regular python you shouldn't need to declare `cdef int i/j`. Unless that explicitly types the values i and j hold i guess

File: test/test_functions.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import testindicators
```
Comment (adrain-cb) [inline test/test_functions.py:2]: What is testindicators

File: test/test_functions.py [hunk 4]
```diff
@@ -36,4 +36,19 @@ def testFIBONACCI(self):
...
+        print(actual)
+        self.assertEqual(expected, actual[0])
+
+    def testEMA(self):
+        vals = np.array([1.9, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0 ,8.0])
+        expected = np.array([1.0, 1.769230842590332, 2.6546761989593506, 3.604093313217163, 4.583608150482178, 5.575806140899658, 6.5729594230651855, 7.571954250335693])
+        actual = np.array(testindicators.EMA(vals, 7))
+        print("actual", actual) 
+        print("expected", expected  )
+        self.assertEqual(expected[-1], actual[-1])
```
Comment (adrain-cb) [inline test/test_functions.py:54]: Noice

File: src/indicators.pyx (hunk 2 above)
Comment (ethenotethan) [reply src/indicators.pyx]: i has to be an int for this instance since j iterates from i to 0

File: test/test_functions.py (hunk 3 above)
Comment (ethenotethan) [reply test/test_functions.py:2]: Testing module, same as indicators but strictly for testing.. it’s built in the unit.sh script before tests are ran

**Pull request merged**